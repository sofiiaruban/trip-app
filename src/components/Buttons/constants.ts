export enum ButtonColor {
  PRIMARY = 'primary',
  BASIC = 'basic'
}

export enum ButtonType {
  BUTTON = 'button',
  SUBMIT = 'submit',
  RESET = 'reset'
}
