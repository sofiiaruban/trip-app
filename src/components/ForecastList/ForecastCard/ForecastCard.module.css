.card {
  display: flex;
  flex-direction: column;
  gap: 1rem;
  color: #ffffff;
  box-sizing: border-box;

  &.big {
    justify-content: center;
    align-items: center;
    background-color: rgb(19, 35, 52);
    width: 100%;
    height: 100%;
    padding: 0.5rem;
  }
}
.day-big {
  font-size: 24px;
  font-weight: 600;
}
.day-small {
  color: rgb(44, 43, 43);
}
.weatherInfo-big {
  display: flex;
  gap: 0.5rem;
  @media (max-width: 790px) {
    flex-wrap: wrap;
    justify-content: center;
  }
}

.weatherInfo-small {
  display: flex;
  flex-direction: column;
  gap: 0.5rem;
}
.degree-big {
  font-size: 34px;
  font-weight: 400;
  margin-bottom: 0;
}
.degree-small {
  color: rgb(63, 62, 62);
}
.img-small {
  height: 50px;
  width: 50px;
}
.img-big {
  width: 100px;
  height: 100px;
}
