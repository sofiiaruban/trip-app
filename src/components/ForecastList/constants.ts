export enum ForecastCardType {
  SMALL = 'small',
  BIG = 'big'
}
