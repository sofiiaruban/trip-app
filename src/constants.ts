export enum CityList {
  NEW_YORK = 'New York',
  LONDON = 'London',
  KYIV = 'Kyiv'
}

export enum CityImg {
  NEW_YORK = 'src/assets/newyork.jpeg',
  LONDON = 'src/assets/london.jpg',
  KYIV = 'src/assets/kyiv.jpg'
}
