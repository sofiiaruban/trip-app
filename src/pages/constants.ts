export const MAIN_PAGE_TEST_ID = 'mainPage'
