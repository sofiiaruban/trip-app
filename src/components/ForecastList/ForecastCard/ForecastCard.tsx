import { FC } from 'react'
import getDayFromDate from '@app/helpers/getDayFromDate'
import { ForecastProps } from '@app/types/types'
import styles from './Forecast.module.css'
import { ForecastCardType } from '../constants'
import classnames from 'classnames'

const ForecastCard: FC<ForecastProps> = ({
  date,
  iconSrc,
  iconName,
  degree,
  children,
  type = ForecastCardType.SMALL
}) => {
  const day = getDayFromDate(date)
  const cardClasses = classnames(styles.card, styles[type])

  return (
    <article className={cardClasses}>
      <p className={styles[`day-${type}`]}>{day}</p>
      <div className={styles[`weatherInfo-${type}`]}>
        <img src={iconSrc} alt={iconName} />
        <p className={styles[`degree-${type}`]}>{degree}</p>
      </div>
      {children}
    </article>
  )
}

export default ForecastCard
