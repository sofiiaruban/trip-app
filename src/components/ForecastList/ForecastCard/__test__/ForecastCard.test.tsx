
import { render } from '@testing-library/react';
import ForecastCard from '../ForecastCard';
import { ForecastCardType } from '../../constants';

const commonProps = {
  date: '2024-02-18',
  iconSrc: 'mock-icon-src',
  iconName: 'mock-icon-name',
  degree: "25",
  children: <div>Mock children</div>,
  type: ForecastCardType.SMALL
};

test('ForecastCard matches snapshot', () => {

  const { container } = render(
   <ForecastCard {...commonProps} />
  )

  expect(container).toMatchSnapshot()
});
