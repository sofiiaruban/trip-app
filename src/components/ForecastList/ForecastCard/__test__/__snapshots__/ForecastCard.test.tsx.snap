// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastCard matches snapshot 1`] = `
<div>
  <article
    class="card small"
  >
    <p
      class="day-small"
    >
      Sunday
    </p>
    <div
      class="weatherInfo-small"
    >
      <img
        alt="mock-icon-name"
        class="img-small"
        src="mock-icon-src"
      />
      <p
        class="degree-small"
      >
        25
         
        <span>
          Â°F
        </span>
      </p>
    </div>
    <div>
      Mock children
    </div>
  </article>
</div>
`;
